admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/service/impl/DeviceServiceImpl.java

```diff
@@ -856,6 +855,24 @@ public StatusResponseDto updateDeviceStatus(String id, boolean isActive) {
 			throw new DataNotFoundException(DeviceErrorCode.DEVICE_NOT_EXISTS_EXCEPTION.getErrorCode(),
 					DeviceErrorCode.DEVICE_NOT_EXISTS_EXCEPTION.getErrorMessage());
 		}
+		if(isActive) {
+			deviceSpecificationList = deviceSpecificationRepository.findByIdAndIsDeletedFalseorIsDeletedIsNull(devices.get(0).getDeviceSpecId());
```

Comment:
Should activation and deactivation of the device be dependent on device-specification?

Reply:
activation of  a device should depend on device specification right anusha, if we deactivated the device specification that means we no longer wants to use devices with that specification right anusha.